### PR TITLE
Fix nvme basic testcase for FreeBSD

### PR DESCRIPTION
**Pull request by dsrivastavv**

### CODE REVIEW

File: lisa/features/nvme.py [hunk 1]
```diff
@@ -43,32 +30,13 @@ def enabled(self) -> bool:
...
 
     def get_devices(self) -> List[str]:
-        devices_list = []
-        self._get_device_from_ls()
-        for row in self._ls_devices.splitlines():
-            matched_result = self._device_pattern.match(row)
-            if matched_result:
-                devices_list.append(matched_result.group("device_name"))
-        return devices_list
+        return self._node.tools[Ls].get_nvme_devices()
```
Comment (squirrelsc) [inline lisa/features/nvme.py]: `Ls` should focus it's own logic on "list files". The nvme logic shouldn't be put in ls.

File: lisa/tools/lsblk.py [hunk 2]
```diff
@@ -10,13 +10,15 @@
 from lisa.util import LisaException, find_patterns_groups_in_lines
 
 
-def _get_size_in_bytes(size: int, size_unit: str) -> int:
+def _get_size_in_bytes(size: float, size_unit: str) -> int:
+    if size_unit == "T":
+        return int(size * 1024 * 1024 * 1024 * 1024)
```
Comment (squirrelsc) [inline lisa/tools/lsblk.py:15]: Why convert it to int?

File: lisa/tools/lsblk.py [hunk 3]
```diff
@@ -246,7 +248,7 @@ def find_mountpoint_by_volume_name(
 class BSDLsblk(Lsblk):
     # da1p1          0:103  12G freebsd-ufs                                       - /mnt/resource   # noqa: E501
     _ENTRY_REGEX = re.compile(
-        r"\s*(?P<name>\S+)\s+\d+:\d+\s+(?P<size>\d+)(?P<size_unit>\w+)\s+"
+        r"\s*(?P<name>\S+)\s+\d+:\d+\s+(?P<size>\d+|\d+.\d+)(?P<size_unit>\w+)\s+"
```
Comment (squirrelsc) [inline lisa/tools/lsblk.py:252]: Add examples above.

File: lisa/tools/nvmecli.py [hunk 4]
```diff
@@ -87,3 +95,39 @@ def support_device_format(self, device_name: str) -> bool:
...
+        namespaces_cli = []
+        nvme_list = self.run("list", shell=True, sudo=True)
+        for row in nvme_list.stdout.splitlines():
+            matched_result = self._namespace_cli_pattern.match(row)
+            if matched_result:
+                namespaces_cli.append(matched_result.group("namespace"))
+        return namespaces_cli
+
+
+class BSDNvmecli(Nvmecli):
```
Comment (squirrelsc) [inline lisa/tools/nvmecli.py:109]: Does it use the same installation as Linux?

Comment (dsrivastavv) [reply lisa/tools/nvmecli.py:109]: It's preinstalled in BSD

Comment (squirrelsc) [reply lisa/tools/nvmecli.py:109]: You can set can_install to false to prevent installation logic reused.

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**